pkg/webui/lib/diff.js

```diff
@@ -0,0 +1,28 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const shallowDiff = function (initial, changed) {
```

Comment:
Note: this will only work for primitive values, not arrays or nested values that [formik also supports](https://jaredpalmer.com/formik/docs/guides/arrays). Not relevant now, but good to keep in mind. Might make sense to use a more thorough external library here later.

Reply:
this is why the functions is called `shallowDiff`. i have no forms with arrays or nested values within this pr.